### DISCUSSION TITLE
Custom firmware for Philips Hue SML001

### DISCUSSION THREAD

**Discussion by LilleGe** [Koenkk/zigbee2mqtt#24725] [Devices]
I have been strugglin with some update issues related to the SML001 Philips Hue Motion Sensor occupancy.
When it detects motion, the state will remain for a minimum amount of seconds, even if no motion is detected, before it resets to unoccupied.

I have read that there is no real workaround for this, as it is implemented in the Philips firmware.. 

So while I was searching for a custom firmware, I have found several for Philips light bulbs but it seems that for the Sensor, the Firmware is unavailable and no one has ever created a custom one.

I have [read](https://slava-moskvin.medium.com/extracting-firmware-every-method-explained-e94aa094d0dd), it might be possible to "read" the firmware or the source where it is coming from, in an attempt to copy the encrypted firmware, while a new version is being flashed with OTA, but I am alread on the latest version and downgrading also does not seem possible since there is no firmware available for the SML001.

Long story short: Is there any way to get my hands on a custom firmware or extract it somehow (without accessing the chip directly), in order to somehow reverse engineer it to a custom firmware, that I can implement with a function to reset the occupancy instantly or after configured time?

**Comment by aburow** [comment]
I've having an issue in ZHA but I don't think that the issue is in the integration. 
Devices with file version: 1107315168 or software version 6.1.0.18912 are working fine on my network.
If they are updated to file version: 1107323831 or software version 6.1.1.27575 then I'm having issues.

The question I need to try and answer is - how do I downgrade?
